tests/rms/test_generate_petro_jobs_for_field_update.py

```diff
@@ -0,0 +1,404 @@
+
+"""
+
+import contextlib
+import copy
+import filecmp
+import shutil
+from os.path import isdir
+from pathlib import Path
+
+import pytest
+import xtgeo
+
+with contextlib.suppress(ImportError):
+    import roxar  # type: ignore
+    import roxar.jobs  # type: ignore
+
+from fmu.tools.rms.generate_petro_jobs_for_field_update import (
+    check_rms_project,
+    create_new_petro_job_per_facies,
+    define_new_variable_names_and_correlation_matrix,
+    read_specification_file,
+    write_petro_job_to_file,
+)
+
+# ======================================================================================
+# settings to create RMS project!
+REMOVE_RMS_PROJECT_AFTER_TEST = True
+
+TMPD = Path("TMP")
+TMPD.mkdir(parents=True, exist_ok=True)
+
+
+PROJNAME = "tmp_project_generate_petro_jobs.rmsxxx"
+PRJ = str(TMPD / PROJNAME)
+RESULTDIR = TMPD / "jobs"
+RESULTDIR.mkdir(parents=True, exist_ok=True)
+
+REFERENCE_DIR = Path("rms/generate_jobs_testdata")
+CONFIG_FILE_ORIGINAL_SINGLE_ZONE_JOB = (
+    REFERENCE_DIR / "generate_original_single_zone_job.yml"
+)
+CONFIG_FILE_ORIGINAL_MULTI_ZONE_JOB = (
+    REFERENCE_DIR / "generate_original_multi_zone_job.yml"
+)
+CONFIG_FILE_SINGLE_ZONE = REFERENCE_DIR / "generate_petro_single_zone_jobs.yml"
+CONFIG_FILE_MULTI_ZONE = REFERENCE_DIR / "generate_petro_multi_zone_jobs.yml"
+JOB_TYPE = "Petrophysical Modeling"
+
+
+REFERENCE_FILES_SINGLE_ZONE_GRID = [
+    REFERENCE_DIR / "F1_single_zone_petro.txt",
+    REFERENCE_DIR / "F2_single_zone_petro.txt",
+    REFERENCE_DIR / "F3_single_zone_petro.txt",
+]
+REFERENCE_FILES_MULTI_ZONE_GRID = [
+    REFERENCE_DIR / "F1_multi_zone_petro.txt",
+    REFERENCE_DIR / "F2_multi_zone_petro.txt",
+    REFERENCE_DIR / "F3_multi_zone_petro.txt",
+]
+
+OWNER_STRING_SINGLE_ZONE = ["Grid models", "SingleZoneBox", "Grid"]
+OWNER_STRING_MULTI_ZONE = ["Grid models", "MultiZoneBox", "Grid"]
+
+
+@pytest.mark.parametrize(
+    "original_variable_names, facies_name, new_variable_names_input, "
+    "original_Lcorr_mat, reference_var_names, reference_corr_mat",
+    [
+        (
+            ["phit", "vsh", "klogh", "vphyl"],
+            "F1",
+            ["F1_klogh", "F1_phit"],
+            [[], [0.2], [0.3, 0.4], [0.5, 0.6, 0.7]],
+            ["F1_phit", "F1_klogh"],
+            [[], [0.3]],
+        ),
+        (
+            ["phit", "vsh", "klogh", "vphyl"],
+            "F2",
+            ["F2_vphyl", "F2_vsh", "F2_klogh"],
+            [[], [0.2], [0.3, 0.4], [0.5, 0.6, 0.7]],
+            ["F2_vsh", "F2_klogh", "F2_vphyl"],
+            [[], [0.4], [0.6, 0.7]],
+        ),
+        (
+            ["phit", "vsh", "klogh", "vphyl"],
+            "F3",
+            ["F3_klogh"],
+            [[], [0.2], [0.3, 0.4], [0.5, 0.6, 0.7]],
+            ["F3_klogh"],
+            [],
+        ),
+    ],
+)
+def test_define_new_variable_names_and_correlation_matrix(
+    original_variable_names,
+    facies_name,
+    new_variable_names_input,
+    original_Lcorr_mat,
```

Comment:
Avoid uppercase letters in variables in python (PEP8 compliance), e.g.
`original_Lcorr_mat  ->  original_lowercorr_matrix`